pkg/generate/app/cmd/newapp.go

```diff
@@ -269,8 +269,12 @@ func (c *AppConfig) validateBuilders(components app.ComponentReferences) error {
 }
 
 func validateEnforcedName(name string) error {
-	if reasons := validation.ValidateServiceName(name, false); len(reasons) != 0 && !app.IsParameterizableValue(name) {
-		return fmt.Errorf("invalid name: %s. Must be an a lower case alphanumeric (a-z, and 0-9) string with a maximum length of 24 characters, where the first character is a letter (a-z), and the '-' character is allowed anywhere except the first or last character.", name)
+	// The limiting factor on name length is that name may be substituted into
+	// imageapi.ContainerImageEntrypointAnnotationFormatKey, whose part after
+	// the '/' character cannot exceed 63 characters.
```

Comment:
I'm a little bit worried that this might be the limiting factor today but not tomorrow (fragile, as you implied).

But do we know where the 24 character limit assumption even came from?  why was it reported as a bug that we're not limiting the name to 24 characters?